sentry-core/src/main/java/io/sentry/core/SendCachedEventFireAndForgetIntegration.java

```diff
@@ -20,20 +20,21 @@ public void register(@NotNull IHub hub, @NotNull SentryOptions options) {
     File outbox = new File(cachedDir);
 
     try {
-      Executors.callable(
-              () -> {
-                try {
-                  sender.sendCachedFiles(outbox);
-                } catch (Exception e) {
-                  logIfNotNull(
-                      options.getLogger(),
-                      SentryLevel.ERROR,
-                      "Failed trying to send cached events at %s",
-                      e,
-                      outbox);
-                }
-              })
-          .call();
+      ExecutorService es = Executors.newFixedThreadPool(1);
```

Comment:
you may use `newSingleThreadExecutor` which is almost the same, but if a failure happens, a new thread takes place